### PR TITLE
global: update query state from response

### PR DESCRIPTION
**Pull request by zzacharo**

### CODE REVIEW

File: src/lib/api/contrib/elasticsearch/ESResponseSerializer.js [hunk 1]
```diff
@@ -16,10 +16,12 @@ export class ESResponseSerializer {
...
   serialize(payload) {
+    const { aggregations, hits, ...extras } = payload;
     return {
-      aggregations: payload.aggregations || {},
-      hits: payload.hits.hits.map(hit => hit._source),
-      total: payload.hits.total.value,
+      aggregations: aggregations || {},
+      hits: hits.hits.map((hit) => hit._source),
+      total: hits.total.value,
+      extras: extras,
```
Comment (ntarocco) [inline src/lib/api/contrib/elasticsearch/ESResponseSerializer.js:24]: this `extras` is really a black hole, an open door to anything :(

File: src/lib/components/ReactSearchKit/ReactSearchKit.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ export class ReactSearchKit extends Component {
         ? props.urlHandlerApi.customHandler ||
           new UrlHandlerApi(props.urlHandlerApi.overrideConfig)
         : null,
+      updateQueryStateFromResponse: props.updateQueryStateFromResponse,
```
Comment (ntarocco) [inline src/lib/components/ReactSearchKit/ReactSearchKit.js:29]: IMHO, this should be handled by the response serializer only, so an option there and not here at the RSK app. WDYT?

**Review by ntarocco — COMMENTED** [review]
Since you want to modify the `queryState` on response, why not:
* keep the `extras` only at the level of the Invenio response serializer. If you use another serializer, e.g. Elasticsearch, probably you will never use this
* change the queryState in the response serializer. It is not idea, but at least it does not "pollute" the entire app.

Let me know what you think :)

File: src/lib/components/ReactSearchKit/ReactSearchKit.js (hunk 2 above)
Comment (zzacharo) [reply src/lib/components/ReactSearchKit/ReactSearchKit.js:29]: Hmm but the serializer doesn't know how to update the state...I need to modify the way the `executeQuery` action is working...what did you have in mind?

File: src/lib/api/contrib/elasticsearch/ESResponseSerializer.js (hunk 1 above)
Comment (zzacharo) [reply src/lib/api/contrib/elasticsearch/ESResponseSerializer.js:24]: This, I just did it for completeness to comply with the Invenio one...I am fine with omitting it :). The `extras` is not mandatory!

File: src/lib/api/contrib/invenio/InvenioSearchApi.js [hunk 3]
```diff
@@ -78,9 +81,18 @@ export class InvenioSearchApi {
...
+    response = this.responseSerializer.serialize(response.data);
+    const newQueryState = updateQueryState(
+      stateQuery,
+      response.extras,
+      STORE_KEYS
+    );
+    if (!_isEmpty(newQueryState)) {
+      response.newQueryState = newQueryState;
+    }
+    return response;
```
Comment (ntarocco) [inline src/lib/api/contrib/invenio/InvenioSearchApi.js:96]: delete the `extras` before? (double check if it works if not set!)
```suggestion
    delete response.extras
    return response;
```

File: src/lib/state/selectors/query.js [hunk 4]
```diff
@@ -120,14 +120,14 @@ export const updateQueryFilters = (queryFilter, stateFilters) => {
...
-    filter => (tempStateFilters = updateFilter(filter, tempStateFilters))
+    (filter) => (tempStateFilters = updateFilter(filter, tempStateFilters))
   );
   return tempStateFilters;
 };
 
 export const updateQueryState = (oldState, newState, storeKeys) => {
   let pickedState = _pick(newState, storeKeys);
-  if ('filters' in pickedState) {
+  if ('filters' in pickedState && !_isEmpty(pickedState.filters)) {
```
Comment (ntarocco) [inline src/lib/state/selectors/query.js:130]: maybe the opposite?
```suggestion
  if (!_isEmpty(pickedState.filters) && 'filters' in pickedState) {
```

**Review by ntarocco — COMMENTED** [review]
I prefer this version! :)

**Review by ntarocco — APPROVED** [review]

**Pull request merged**